src/payment/strategies/googlepay/googlepay-payment-strategy.ts

```diff
@@ -0,0 +1,371 @@
+import { RequestSender } from '@bigcommerce/request-sender';
+import Response from '@bigcommerce/request-sender/lib/response';
+
+import { BillingAddressActionCreator } from '../../../billing';
+import { BillingAddressUpdateRequestBody } from '../../../billing';
+import CheckoutStore from '../../../checkout/checkout-store';
+import { CheckoutActionCreator } from '../../../checkout/index';
+import InternalCheckoutSelectors from '../../../checkout/internal-checkout-selectors';
+import {
+    InvalidArgumentError,
+    MissingDataError,
+    MissingDataErrorType,
+    NotInitializedErrorType,
+    StandardError
+} from '../../../common/error/errors/index';
+import NotInitializedError from '../../../common/error/errors/not-initialized-error';
+import { toFormUrlEncoded } from '../../../common/http-request';
+import { bindDecorator as bind } from '../../../common/utility';
+import {
+    OrderActionCreator,
+    OrderRequestBody
+} from '../../../order/index';
+import { RemoteCheckoutSynchronizationError } from '../../../remote-checkout/errors';
+import ConsignmentActionCreator from '../../../shipping/consignment-action-creator';
+import {
+    PaymentActionCreator,
+    PaymentMethodActionCreator,
+    PaymentStrategyActionCreator
+} from '../../index';
+import Payment from '../../payment';
+import PaymentMethod from '../../payment-method';
+import {
+    PaymentInitializeOptions,
+    PaymentRequestOptions
+} from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import {
+    default as mapGooglePayAddressToRequestAddress,
+    ButtonColor,
+    ButtonType,
+    EnvironmentType,
+    GooglePaymentsError,
+    GooglePaymentData,
+    GooglePayAddress,
+    GooglePayClient,
+    GooglePayInitializer,
+    GooglePayIsReadyToPayResponse,
+    GooglePayPaymentDataRequestV1,
+    GooglePayPaymentOptions, GooglePaySDK,
+    PaymentSuccessPayload,
+    TokenizePayload
+} from './googlepay';
+import GooglePayPaymentInitializeOptions from './googlepay-initialize-options';
+import GooglePayScriptLoader from './googlepay-script-loader';
+
+export default class GooglePayPaymentStrategy extends PaymentStrategy {
```

Comment:
Since most of the logic in this class is going to be reused in the `CustomerStrategy` and the `CheckoutButtonStrategy`, it might be a good idea to split this class in 2. A payment processor and the strategy.